crates/networking/p2p/src/network.rs

```diff
@@ -480,4 +750,84 @@ mod tests {
             }
         });
     }
+
+    #[test]
+    fn test_peer_table_lifecycle() {
+        initialize_network_spec();
+
+        let tokio_runtime = Runtime::new().unwrap();
+        let mut network1 = tokio_runtime
+            .block_on(create_network(
+                "127.0.0.1".parse().unwrap(),
+                9300,
+                9301,
+                vec![],
+                true,
+                vec![],
+            ))
+            .unwrap();
+
+        let mut network2 = tokio_runtime
+            .block_on(create_network(
+                "127.0.0.1".parse().unwrap(),
+                9302,
+                9303,
+                vec![],
+                true,
+                vec![],
+            ))
+            .unwrap();
+
+        let addr: Multiaddr = "/ip4/127.0.0.1/tcp/9300".parse().unwrap();
+        network2.swarm.dial(addr).unwrap();
+
+        let peer_id_network1 = network1.peer_id();
+        let peer_id_network2 = network2.peer_id();
+
+        tokio_runtime.block_on(async {
+            let network1_poll_task = async {
+                while let Some(ev) = network1.swarm.next().await {
+                    network1.parse_swarm_event(ev);
+                    if matches!(
+                        network1.cached_peer(&peer_id_network2),
+                        Some(row) if row.state == ConnectionState::Connected && row.direction == Direction::Inbound
+                    ) {
+                        break;
+                    }
+                }
+            };
+
+            let network2_poll_task = async {
+                while let Some(ev) = network2.swarm.next().await {
+                    network2.parse_swarm_event(ev);
+                    if matches!(
+                        network2.cached_peer(&peer_id_network1),
+                        Some(row) if row.state == ConnectionState::Connected && row.direction == Direction::Outbound
+                    ) {
+                        break;
+                    }
+                }
+            };
+
+            tokio::time::timeout(
+                std::time::Duration::from_secs(10),
+                futures::future::join(network1_poll_task, network2_poll_task),
+            )
+            .await
+            .expect("peer-table not updated in time");
+        });
+
+        let row1 = network1
+            .cached_peer(&peer_id_network2)
+            .expect("network1 row exists");
+        let row2 = network2
```

Comment:
row what? peer_from_network_1? What is a row, the variable name should be more informative